main.py

```diff
@@ -91,12 +94,20 @@ def debug_gql(self) -> int:
             return logging.NOTSET
 
 
+    try:
+        root = tk.Tk()
+        root.overrideredirect(True)
+        root.withdraw()
+        root.iconphoto(
+            True, PhotoImage(master=root, image=Image_module.open(resource_path("pickaxe.ico")))
+        )
+        root.update()
+    except Exception:  # root window doesn't created
```

Comment:
This won't work when trying to use the CLI mode on a machine where a Tk window **can** be created. Again, it'd be best to somehow store the error within the parser, and then display it afterwards, with a window when in normal mode, to console when in CLI mode. You may run into an issue where you won't know where/how to display the error when argument parsing itself fails to tell you that information (duh), but an approach where one checks `sys.stdout` for being connected to something that can accept output (like a console) might be the best compromise here. It can not work after packing the application into PyInstaller, by possibly giving you a false reading of some kind of output being present, but should still be worth a try.